rest/admin_api.go

```diff
@@ -133,10 +135,87 @@ func (h *handler) handleGetConfig() error {
 	return nil
 }
 
-// Put admin config info
 func (h *handler) handlePutConfig() error {
-	h.writeStatus(http.StatusOK, "updated")
-	return nil
+
+	type FileLoggerPutConfig struct {
+		Enabled *bool `json:"enabled,omitempty"`
+	}
+
+	type ConsoleLoggerPutConfig struct {
+		LogLevel *base.LogLevel `json:"log_level,omitempty"`
+		LogKeys  []string       `json:"log_keys,omitempty"`
+	}
+
+	// Probably need to make our own to remove log file path / redaction level
+	type ServerPutConfig struct {
+		Logging struct {
+			Console *ConsoleLoggerPutConfig `json:"console,omitempty"`
+			Error   FileLoggerPutConfig     `json:"error,omitempty"`
+			Warn    FileLoggerPutConfig     `json:"warn,omitempty"`
+			Info    FileLoggerPutConfig     `json:"info,omitempty"`
+			Debug   FileLoggerPutConfig     `json:"debug,omitempty"`
+			Trace   FileLoggerPutConfig     `json:"trace,omitempty"`
+			Stats   FileLoggerPutConfig     `json:"stats,omitempty"`
+		} `json:"logging"`
+	}
+
+	body, err := h.readBody()
+	if err != nil {
+		return err
+	}
+
+	var config ServerPutConfig
+	d := base.JSONDecoder(bytes.NewBuffer(body))
+	d.DisallowUnknownFields()
+	err = base.WrapJSONUnknownFieldErr(d.Decode(&config))
+	if err != nil {
+		if pkgerrors.Cause(err) == base.ErrUnknownField {
+			return base.HTTPErrorf(http.StatusBadRequest, "Unknown config field specified: %v", err)
```

Comment:
As discussed, we should present this error as an "Unable to configure given options at runtime" type error, rather than an unknown field error.

Unknown fields here are an implementation detail to avoid us having to look over the entire existing StartupConfig struct.